lms/djangoapps/course_home_api/dates/v1/serializers.py

```diff
@@ -27,10 +27,11 @@ class DateSummarySerializer(serializers.Serializer):
     first_component_block_id = serializers.SerializerMethodField()
 
     def get_learner_has_access(self, block):
-        learner_is_full_access = self.context.get('learner_is_full_access', False)
-        block_is_verified = (getattr(block, 'contains_gated_content', False) or
-                             isinstance(block, VerificationDeadlineDate))
-        return (not block_is_verified) or learner_is_full_access
+        if isinstance(block, VerificationDeadlineDate):
+            # This date block isn't an assignment, so doesn't have contains_gated_content set for it
+            return self.context.get('learner_is_full_access', False)
+
+        return not getattr(block, 'contains_gated_content', False)
```

Comment:
This was just cleanup - after I realized that `contains_gated_content` already looks at whether learner is full access, I broke these two paths out to hopefully make it clearer why we look at both.